benches/src/erc1155_supply.rs

```diff
@@ -0,0 +1,75 @@
+use alloy::{
+    network::{AnyNetwork, EthereumWallet},
+    primitives::Address,
+    providers::ProviderBuilder,
+    sol,
+    sol_types::SolCall,
+    uint,
+};
+use e2e::{receipt, Account};
+
+use crate::{
+    report::{ContractReport, FunctionReport},
+    CacheOpt,
+};
+
+sol!(
+    #[sol(rpc)]
+    contract Erc1155Supply {
+        function totalSupply(uint256 id) external view returns (uint256);
+        function totalSupply() external view returns (uint256);
+        function exists(uint256 id) external view returns (bool);
+    }
+);
+
+pub async fn bench() -> eyre::Result<ContractReport> {
+    let reports = run_with(CacheOpt::None).await?;
+    let report = reports
+        .into_iter()
+        .try_fold(ContractReport::new("Erc1155Supply"), ContractReport::add)?;
+
+    let cached_reports = run_with(CacheOpt::Bid(0)).await?;
+    let report = cached_reports
+        .into_iter()
+        .try_fold(report, ContractReport::add_cached)?;
+
+    Ok(report)
+}
+
+pub async fn run_with(
+    cache_opt: CacheOpt,
+) -> eyre::Result<Vec<FunctionReport>> {
+    let alice = Account::new().await?;
+    let alice_wallet = ProviderBuilder::new()
+        .network::<AnyNetwork>()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(alice.signer.clone()))
+        .on_http(alice.url().parse()?);
+
+    let contract_addr = deploy(&alice, cache_opt).await?;
+
+    let contract = Erc1155Supply::new(contract_addr, &alice_wallet);
+
+    let token = uint!(1_U256);
+
+    // IMPORTANT: Order matters!
+    use Erc1155Supply::*;
+    #[rustfmt::skip]
+    let receipts = vec![
```

Comment:
I believe you should mint some tokens to make benches more accurate. Right now you do not have any tokens minted so `total_supply(token)` or `exists(token)` may have slightly different gas usage.

Reply:
As a bonus, we also get to bench the extended `_update` fn